### PR TITLE
Updates sms input to dropdown. Allows multiple country codes.

### PR DESCRIPTION
**Pull request by Nathan-Notifi**
Creates a dropdown for the SMS Input.

- Adds new phone input dropdown to iterate based off country codes.

image

dark mode: 
image

Demo:
https://user-images.githubusercontent.com/105258726/193683673-b5adb981-3a0e-41ca-beb7-034df013a26e.mov

### CODE REVIEW

File: packages/notifi-react-card/lib/components/NotifiSmsInput.tsx [hunk 1]
```diff
@@ -13,26 +14,97 @@ export type NotifiSmsInputProps = Readonly<{
...
+      if (field === 'dialCode') {
+        setPhoneNumber(event.target.value + phoneValues.baseNumber);
+      } else {
+        setPhoneNumber(phoneValues.dialCode + event.target.value);
+      }
+    };
+
   return (
     <div className={clsx('NotifiSmsInput__container', classNames?.container)}>
+      <div className={clsx('NotifiSmsInput__dropdownContainer')}>
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/components/NotifiSmsInput.tsx:92]: let's add a className to the props as well so people can override it if necessary

File: packages/notifi-react-card/lib/components/NotifiSmsInput.tsx [hunk 2]
```diff
@@ -13,26 +14,97 @@ export type NotifiSmsInputProps = Readonly<{
...
+        <select
+          value={phoneValues.dialCode}
+          onChange={handleChange('dialCode')}
+          className={clsx('NotifiSmsInput__dropdownSelect')}
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/components/NotifiSmsInput.tsx:96]: ditto re: adding a value to classNames prop

File: packages/notifi-react-card/lib/components/subscription/defaults.css [hunk 3]
```diff
@@ -177,3 +179,35 @@
...
+  color: var(--notifi-font-color);
+  background-color: var(--notifi-input-background);
+}
+
+.NotifiSmsInput__dropdownSelect::after {
+  content: '';
+  width: 0.8em;
+  height: 0.5em;
+  background-color: var(--notifi-input-background);
+  clip-path: polygon(100% 0%, 0 0%, 50% 100%);
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/components/subscription/defaults.css:207]: ooh what is this

File: packages/notifi-react-card/tsconfig.json [hunk 4]
```diff
@@ -2,5 +2,6 @@
   "extends": "../../tsconfig.json",
   "compilerOptions": {
     "jsx": "react",
+    "lib": ["ES2021", "DOM"]
```
Comment (juni-notifi) [inline packages/notifi-react-card/tsconfig.json:5]: what effect does this have on tsup? just curious if ES2021 here will affect our compatibility (and by how much)

**Review by juni-notifi — COMMENTED** [review]
small points of feedback~ lmk when you've had a chance to look!

File: packages/notifi-react-card/lib/components/subscription/defaults.css (hunk 3 above)
Comment (Nathan-Notifi) [reply packages/notifi-react-card/lib/components/subscription/defaults.css:207]: oh shoot, forgot to take this out.

File: packages/notifi-react-card/tsconfig.json (hunk 4 above)
Comment (Nathan-Notifi) [reply packages/notifi-react-card/tsconfig.json:5]: I added support for ES2021 due to it impacting the `replaceAll` method. 

Would you recommend using something from here instead?
https://dmitripavlutin.com/replace-all-string-occurrences-javascript/

Comment (Nathan-Notifi) [reply packages/notifi-react-card/tsconfig.json:5]: does `tsup` stand for typescripti support?

Comment (juni-notifi) [reply packages/notifi-react-card/tsconfig.json:5]: Screen Shot 2022-10-04 at 11 00 35 AM

93% honestly feels good enough, particularly in the world of crypto where most users upgrade browsers often (for security purposes too)

I guess the main alternative would be a string.replace with regex?

**Review by juni-notifi — COMMENTED** [review]
Would you mind double-checking that `useSubscribe` won't try to make invalid phone targets

File: package.json [hunk 5]
```diff
@@ -43,5 +43,8 @@
     "packages/notifi-node-sample",
     "packages/notifi-react-card",
     "packages/notifi-react-hooks"
-  ]
+  ],
+  "dependencies": {
+    "libphonenumber-js": "^1.10.13"
+  }
```
Comment (juni-notifi) [inline package.json:49]: just realized this is inside the root package.json

probably want to move this to `notifi-react-card`

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts [hunk 6]
```diff
@@ -183,13 +183,8 @@ export const useNotifiSubscribe: () => Readonly<{
...
     let finalPhoneNumber = null;
-    if (inputPhoneNumber !== '') {
-      if (inputPhoneNumber.startsWith('+')) {
-        finalPhoneNumber = inputPhoneNumber;
-      } else {
-        // Assume US for now
-        finalPhoneNumber = `+1${inputPhoneNumber}`;
-      }
+    if (isValidPhoneNumber(inputPhoneNumber)) {
+      finalPhoneNumber = inputPhoneNumber;
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:187]: do we no longer need the `+` logic?

**Review by juni-notifi — APPROVED** [review]

File: packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts (hunk 6 above)
Comment (Nathan-Notifi) [reply packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts:187]: yeah, no longer need it.
[
    "+1",
    "+61",
    "+43",
    "+32",
    "+55",
    "+45",
    "+358",
    "+33",
    "+49",
    "+852",
    "+36",
    "+354",
    "+60",
    "+47",
    "+63",
    "+48",
    "+351",
    "+65",
    "+82",
    "+34",
    "+46",
    "+41",
    "+886",
    "+44"
]

**Pull request merged**